homeassistant/components/sensor/haveibeenpwned.py

```diff
@@ -0,0 +1,152 @@
+"""
+Support for haveibeenpwned (email breaches) sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.haveibeenpwned/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+import requests
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (STATE_UNKNOWN, CONF_EMAIL)
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+DATE_STR_FORMAT = "%Y-%m-%d %H:%M:%S"
+USER_AGENT = "Home Assistant HaveIBeenPwned Sensor Component"
+
+SCAN_INTERVAL = 5
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the RESTful sensor."""
+    emails = config.get(CONF_EMAIL)
+
+    data = HaveIBeenPwnedData(emails)
+    data.update = Throttle(timedelta(seconds=SCAN_INTERVAL-1))(data.update)
```

Comment:
the rest service blocks consequent calls it needs a delay between calls, if i was to request all email data in a loop at startup it would fail. So i wanted to add a 5 second delay between requests at startup to make sure the requests don't fail. In the beginning i've set SCAN_INTERVAL to 5 but if i set the throtlle also at 5 secons it was not updating every 5 seconds but every 10 because it's possible the throttle will block it if scan_interval & throttle are set to the same value. This did not happen if i setted the throttle 1 second lower than the scan_interval.

Futher in the code i raise the scan interval to 15 minutes and i set the throttle to scan_interval - 5 seconds for the same reason.

Basically i wanted to achieve this:
at startup to the requests per email with 5 seconds between each request, as soon as each email has gotten it's data increase the delay to 15 minutes. I also raised scan_interval because it does not need to call update so much.

at First i tried another aproach using the sleep command but this blocks everything hass will pause during startup if i use sleep command so i did not want to use sleep. I then tried to figure out a way to use just the update function as normal but make sure only 1 email is requested in each call to update.

Not sure what the best approach should have been in this case then ?